### PR TITLE
docs: update PR and issue templates and enhancements to CONTRIBUTING.MD

### PR DESCRIPTION
**Pull request by fcastill**
## Description

Updated PR and issue templates and enhancements to CONTRIBUTING doc.

## Motivation and Context

Simplify PR and issue templates as well as improving contributing experience by being more specific about our dev workflow and guidelines.

## How Has This Been Tested?

Docs - no testing required

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build/CI related changes
- [x] Other... Please describe: docs

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

```

```

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,25 +1,41 @@
...
 
+## How Has This Been Tested?
 
-**What is the new behavior?**
+Please describe in detail how you tested your changes.
 
+Include details of your testing environment, and the tests you ran to
 
-**Other information**:
+see how your change affects other areas of the code, etc.
```
Comment (apandurangi) [inline .github/PULL_REQUEST_TEMPLATE.md]: Does that sentence is `Say how your change...` ? or is it `see how your change...`.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,25 +1,41 @@
...
-```
+Describe your changes in detail
 
-**What is the current behavior?** (You can also link to an open issue here)
+## Motivation and Context
 
+Why is this change required? What problem does it solve?
+If it fixes an open issue, please link to the issue here.
 
+## How Has This Been Tested?
```
Comment (apandurangi) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
## How has this been tested?
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,60 +1,60 @@
...
-```
+- `yarn` or `yarn install` to make sure your dependencies are up to date
+- `yarn start server` start the server app in development mode
+- `yarn start client` start the client app in development mode
+- `yarn format:check` to validate all code is formatted
+- `yarn format:write` to format all the code
+- `yarn lint` to lint all packages
+- `yarn lint <package-name>` to lint a specific package. For example `yarn lint plugins-flow-client`
+- `yarn test` to run all unit tests in all packages
+- `yarn test <package-name>` to run all unit tests in a specific package. For example `yarn test plugins-flow-client`
```
[the remark this answers is not in the document]
Comment (fcastill) [reply CONTRIBUTING.md]: 🤔 I will indicate we're leveraging nx and link to their docs as an extra, I don't want to complicate the simple use cases.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -91,3 +143,12 @@ yarn run <command>
 
 Flogo Web uses npm/yarn scripts for its build process. Take a look at README.md of the packages as well as at the scripts section
 in the `package.json` of the root project and the subpackages to find other commands available.
+
+## Server Environment
+
+There are a couple of environment variables you can use to control the server behavior dynamically, they're document in the [`.env.example`](./.env.example) file.
+
+To use them make a copy and rename the [`.env.example`](/.env.example) to `.env` and then uncommentm, add or modify
```
Comment (ppaidi) [inline CONTRIBUTING.md]: typo - uncomment*

**Pull request merged**